[Task]: change $this->get() in controllers to use injection

kingjia90 opened the pull request:
## Changes in this pull request  
Resolves #11745

## Additional info  
the `get` shortcut method in controller is deprecating in symfony 5.4
https://symfony.com/blog/new-in-symfony-5-4-controller-changes#deprecated-some-controller-shortcuts

kingjia90 committed:
change $this->get() to method injection
as the get shortcut method in controller is deprecating in symfony 5.4

kingjia90 committed:
adds missing DI constructor for voucherController

kingjia90 committed:
phpdoc fixes

kingjia90 committed:
fixes(?) and changed typo reponseHelper

kingjia90 committed:
fixes setLocale error

dvesh3 commented on bundles/AdminBundle/Controller/AdminController.php:
```diff
@@ -31,6 +31,15 @@
+    /**
+     * @var TokenStorageUserResolver
+     */
+    protected $tokenResolver;
```
please also remove form `getSubscribedServices()` as not needed anymore to load from service container.

dvesh3 commented on bundles/AdminBundle/Controller/AdminController.php:
```diff
@@ -31,6 +31,15 @@
+    /**
+     * @var TranslatorInterface
+     */
+    protected $translator;
```
remove form `getSubscribedServices()`

dvesh3 replied:
remove form `getSubscribedServices()`

dvesh3 commented on bundles/AdminBundle/Controller/Admin/LoginController.php:
```diff
@@ -43,20 +43,19 @@
-        $this->reponseHelper = $responseHelper;
+    public function __construct(
+        protected ResponseHelper $responseHelper,
+        protected PimcoreBundleManager $bundleManager
```
I would define `$bundleManager` in abstract controller.

dvesh3 replied:
I would define `$bundleManager` in abstract controller.

dvesh3 commented on bundles/EcommerceFrameworkBundle/Controller/VoucherController.php:
```diff
@@ -37,6 +37,18 @@
  */
 class VoucherController extends FrontendController implements KernelControllerEventInterface
 {
+    protected $translator;
```
define in abstract controller.

dvesh3 replied:
define in abstract controller.

dvesh3 reviewed (commented):
Should we also replace `$this->container->get()` calls in AdminController? e.g. 

https://github.com/pimcore/pimcore/blob/b2f37a652b19637bf3c260de7333197d087ba2ac/bundles/AdminBundle/Controller/AdminController.php#L185-L189

kingjia90 committed:
further improvements
removed from stuff subscribedservices, moved some services in the abstract class

kingjia90 committed:
added contentawareinterface and fixed editablerenderer type

kingjia90 committed:
DI optional admin serializer

kingjia90 committed:
serialiazer nullable (?)

kingjia90 committed:
revert serialiazer to container->get and revert the nullable

kingjia90 committed:
revert serializer to container->get and revert the nullable

dvesh3 commented on bundles/EcommerceFrameworkBundle/Controller/VoucherController.php:
```diff
@@ -37,13 +37,22 @@
     public static function getSubscribedServices(): array
     {
         $services = parent::getSubscribedServices();
-        $services['translator'] = TranslatorInterface::class;
```
Is it still required to include `TokenStorageUserResolver` service here? if not, then please kick the method.

kingjia90 replied:
I've moved it, but i am not sure if TokenStorageUserResolver should belong the abstract class (which is FrontendController).
This voucherController is for when managing the voucher generation and stats which loads a frontend-insh page in a iframe but it actually requires admin access or so. wdyt?

dvesh3 replied:
I would not move it to FrontendController as the usage is quite specific to VoucherController. It is basically used to set the locale from user for the integrated view to load the correct translations.

kingjia90 replied:
ok, then i will revert last commit

dvesh3 commented on lib/Controller/FrontendController.php:
```diff
@@ -30,18 +32,33 @@
-        $services[EditableRenderer::class] = '?'.EditableRenderer::class;
-
-        return $services;
+        return parent::getSubscribedServices();
```
please remove the method as it is not used anymore to include services.

kingjia90 committed:
moved tokenResolver,
removed getSubscribedServices

kingjia90 committed:
moved tokenResolver back to voucher Controller

dvesh3 commented on lib/Controller/FrontendController.php:
```diff
@@ -79,6 +91,61 @@ public function __set($name, $value)
+     * @required
+     * @param DocumentResolver $documentResolver
+     */
+    public function setDocumentResolver(DocumentResolver $documentResolver): void
```
```suggestion
    public function setDocumentResolver(DocumentResolver $documentResolver)
```
this can be a BC break if implemented in the subclass. e.g. https://github.com/pimcore/portal-engine/blob/2.x/src/Controller/Rest/Api/AbstractRestApiController.php#L149

dvesh3 replied:
not sure if we really introduce these methods in abstract class

kingjia90 committed:
removing return type in abstract class

kingjia90 committed:
Frontendcontroller services back to subscribed service to avoid potential BC + added getters
VoucherController using constructor injection, since it's internal

kingjia90 committed:
fix the resolvers

kingjia90 committed:
apply to AdminController.php

dvesh3 committed:
Revert frontendcontroller injections

dvesh3 commented:
failing tests are not related. Rebase once https://github.com/pimcore/pimcore/pull/12193 is merged.

dvesh3 committed:
Merge branch '10.x' into remove_get_calls

# Conflicts:
#	bundles/AdminBundle/Controller/AdminController.php
